app/policies/api/assignment_policy.rb

```diff
@@ -12,5 +12,9 @@ def submit_file?
     def see_hidden?
       role.instructor? || role.ta? || role.visible_assessments(assessment_id: record.id).exists?
     end
+
+    def index?
+      real_user.is_a?(EndUser) && real_user.roles.exists?
```

Comment:
This policy isn't strict enough. Checking `real_user.roles.exists?` doesn't check whether that user has a role in the current assignment's course. (Remember that this route is nested within `courses/`)